core/controllers/galleries.py

```diff
@@ -121,8 +152,8 @@ def get(self):
 class GalleryHandler(base.BaseHandler):
     """Provides data for the exploration gallery page."""
 
-    def get(self):
-        """Handles GET requests."""
+    def get_data(self):
```

Comment:
Are you trying to do something like get_explorations_list(), perhaps? All the self.values.update() stuff ought to go in the handler; the common function should just return data and be unaware of its surroundings.